internal/provider/kubernetes/controller.go

```diff
@@ -747,6 +744,22 @@ func (r *gatewayAPIReconciler) processSecurityPolicyObjectRefs(
 					"failed to process OIDC SecretRef for SecurityPolicy",
 					"policy", policy, "secretRef", oidc.ClientSecret)
 			}
+
+			var backendRef gwapiv1.BackendObjectReference
+			if oidc.Provider.BackendRef != nil {
+				backendRef = *oidc.Provider.BackendRef
+			}
+			if len(oidc.Provider.BackendRefs) > 0 {
+				backendRef = oidc.Provider.BackendRefs[0].BackendObjectReference
```

Comment:
why arent we inserting all ?

Reply:
this change however breaks functionality, because before we maybe incorrectly processing here, but with processbackendRefs we were still processing all and they would be available in the gateway-api layer.

now since we have limited processing, its going to break those multi backendRef cases

Reply:
cc @zhaohuabing